core/new-gui/src/app/workspace/component/workspace.component.ts

```diff
@@ -163,34 +181,78 @@ export class WorkspaceComponent implements AfterViewInit, OnInit, OnDestroy {
   loadWorkflowWithId(wid: number): void {
     // disable the workspace until the workflow is fetched from the backend
     this.workflowActionService.disableWorkflowModification();
-    this.workflowPersistService
-      .retrieveWorkflow(wid)
-      .pipe(untilDestroyed(this))
-      .subscribe(
-        (workflow: Workflow) => {
-          // enable workspace for modification
-          this.workflowActionService.toggleLockListen(false);
-          this.workflowActionService.enableWorkflowModification();
-          // load the fetched workflow
-          this.workflowActionService.reloadWorkflow(workflow);
-          // clear stack
-          this.undoRedoService.clearUndoStack();
-          this.undoRedoService.clearRedoStack();
-          this.workflowActionService.toggleLockListen(true);
-          this.workflowActionService.syncLock();
-        },
-        () => {
-          // enable workspace for modification
-          this.workflowActionService.enableWorkflowModification();
-          // clear the current workflow
-          this.workflowActionService.reloadWorkflow(undefined);
-          // clear stack
-          this.undoRedoService.clearUndoStack();
-          this.undoRedoService.clearRedoStack();
 
-          this.message.error("You don't have access to this workflow, please log in with an appropriate account");
-        }
-      );
+    // if display particular execution is true
+    if (this.execution_flag && this.execution) {
+      this.workflowExecutionService
+        .displayWorkflowExecution(wid, this.execution.vId)
+        .pipe(untilDestroyed(this))
+        .subscribe(
+          (workflow: Workflow) => {
+            // we need to display the version on the paper but keep the original workflow in the background
+            this.workflowActionService.setTempWorkflow(this.workflowActionService.getWorkflow());
+            // disable persist to DB because it is read only
+            this.workflowPersistService.setWorkflowPersistFlag(false);
+            // disable the undoredo service because reloading the workflow is considered an action
+            this.undoRedoService.disableWorkFlowModification();
+            // enable modidification to reload workflow
+            this.workflowActionService.enableWorkflowModification();
+            // reload the read only workflow version on the paper
+            this.workflowActionService.reloadWorkflow(workflow);
+
+            if (this.execution) {
+              this.workflowWebsocketService.openExecutionWebsocket(
```

Comment:
you are already inside the iff, why do you need to check again?

Reply:
Changed the way how execution is defined. No need to check again now